src/framework/ui/internal/uiconfiguration.cpp

```diff
@@ -315,6 +330,23 @@ void UiConfiguration::synchThemeWithSystemIfNecessary()
     doSetIsDarkMode(platformTheme()->isSystemThemeDark());
 }
 
+void UiConfiguration::synchAccentColorWithSystemIfNecessary()
+{
+    if (QSysInfo::productType() != "osx" || !m_isFollowSystemTheme.val) {
```

Comment:
This check will be performed at runtime, but OS checks are better done at compile time. You can do that using the following:
```c++
#ifdef Q_OS_MAC
    // do macOS specific stuff
#endif
```
This will also fix the unit tests, because currently you are trying to use `MacOSPlatformTheme::getAccentColorIndex` on all platforms, while it only exists on macOS.